src/js/jsx/search/SearchBar.jsx

```diff
@@ -416,15 +429,19 @@ define(function (require, exports, module) {
          * Perform action based on ID
          *
          * @private
-         * @type {Datalist~onChange}
+         * @type {Datalist~onInputChange}
+         * @param {string} id of the selected option
+         * @param {string} searchTerm user input
+         * @return {{dontCloseDialog: boolean=}};
          */
-        _handleChange: function (id) {
-            if (id === null) {
+        _handleChange: function (id, searchTerm) {
+            if (id === null || !id) {
```

Comment:
This can be null (see: datalist.jsx `_handleSelectClick`), so I will change the parameter to optional and only have the `!id` check.